static/js/search.js

```diff
@@ -44,36 +41,42 @@
             +'</div>';
     }
 
-    window.renderBingSearchResults = () => {
-      let urlParams = new URLSearchParams(window.location.search);
-      let searchTerm = urlParams.get("q") || "";
-      let page = urlParams.get("page") || 1;
-      let q = searchTerm;
-      let results = '';
-      let offset = (page - 1) * 10;
-      let ajaxConf = {};
-
-      if (!searchTerm) return;
-
-      ajaxConf.url = 'https://kubernetes-io-search.azurewebsites.net/api/bingsearchproxy';
-      ajaxConf.data =  { q: q, offset: offset };
-      ajaxConf.type = "GET";
-
-      $.ajax(ajaxConf).done(function(res) {
-        if (res.status === 500) {
-          console.log('Server Error');
-          return;
+    window.renderPageFindSearchResults = () => {
+        let urlParams = new URLSearchParams(window.location.search);
+        let searchTerm = urlParams.get("q") || "";
+        let sidebarSearch = document.querySelector('.td-sidebar__search');
+        if (sidebarSearch) {
+            sidebarSearch.remove();
+        }
+        document.getElementById('search').style.display = 'block';
+        pagefind = new PagefindUI({ element: "#search", showImages: false });
+        if (searchTerm) {
+            pagefind.triggerSearch(searchTerm);
         }
 
-        if (res.webPages == null) return; // If no result, 'webPages' is 'undefined'
-        var paginationAnchors = window.getPaginationAnchors(Math.ceil(res.webPages.totalEstimatedMatches / 10));
-        res.webPages.value.map(ob => { results += window.getResultMarkupString(ob); })
-
-        if($('#bing-results-container').length > 0) $('#bing-results-container').html(results);
-        if($('#bing-pagination-container').length > 0) $('#bing-pagination-container').html(paginationAnchors);
-      });
+        document.querySelector("#search input").addEventListener("input", function() {
+            var inputValue = this.value;
+            var queryStringVar = "q";
+            updateQueryString(queryStringVar, inputValue);
+        });
     }
 
+	function updateQueryString(key, value) {
+		var baseUrl = window.location.href.split("?")[0];
+		var queryString = window.location.search.slice(1);
+		var urlParams = new URLSearchParams(queryString);
+
+		if (urlParams.has(key)) {
+			urlParams.set(key, value);
+		} else {
+			urlParams.append(key, value);
+		}
+
+		var newUrl = baseUrl + "?" + urlParams.toString();
+		// Update the browser history (optional)
+		window.history.pushState({}, null, newUrl);
```

Comment:
(nit / doesn't block merge)
A tidier way to do this is only if either:
- the user clicks on a PageFind link
- the user submits the form

To see the problem, edit a search input and then try navigating back.

Reply:
It is currently set to perform this on form submit. This happens automatically every time the search text is changed in the input box. The updated url could be useful in the case the user wants to share the search results.

Reply:
You see the snag, though, right? If you have typed a term like PersistentVolume, and you want to navigate back, you now must click 16 times. Each keystroke adds a history entry.

Reply:
Ok, I've fixed this.